src/features/filter/reducers/toggleClassFilter.thunk.ts

```diff
@@ -0,0 +1,72 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { filterAdapter, simpleSelectFilterById } from "../filter.selectors";
+import { RootState } from "../../../app/store/store";
+import { FilterKey, Filter, ClassFilterValue } from "../types/filter.types";
+import { jdiClassFilterInit } from "../utils/filterSet";
+import { PageObjectId } from "../../pageObjects/types/pageObjectSlice.types";
+import { ElementClass, ElementLibrary } from "../../locators/types/generationClasses.types";
+import { LocalStorageKey } from "../../../common/utils/const";
+
+interface toggleClassFilterPayload {
+  pageObjectId: PageObjectId;
+  library: ElementLibrary;
+  jdiClass: ElementClass;
+  value: boolean;
+}
+
+interface toggleClassFilterReducerPayload {
+  pageObjectId: PageObjectId;
+  newFilterValue: Filter;
+  initialFilter: ClassFilterValue;
+  newFilter: ClassFilterValue;
+}
+
+export const toggleClassFilter = createAsyncThunk(
+  "filter/toggleClassFilter",
+  async (payload: toggleClassFilterPayload, { getState }) => {
+    const { pageObjectId, jdiClass, value, library } = payload;
+
+    const state = getState() as RootState;
+    const newFilterValue = simpleSelectFilterById(state.filters, pageObjectId);
+
+    const savedFilters = JSON.parse(localStorage.getItem(LocalStorageKey.Filter)!);
+    let newFilter;
+    const initialFilter = {
+      ...jdiClassFilterInit(library),
+      ...(savedFilters && savedFilters[library] && savedFilters[library]),
```

Comment:
why do you write `savedFilters[library]` twice?

Reply:
first one is checking, that savedFilters has library property